### PR TITLE
Backport missing features in general ledger from account_jasper_reports

### PR DESCRIPTION
**Pull request by raimonesteve**
#159087

### CODE REVIEW

File: general_ledger/general_ledger.py [hunk 1]
```diff
@@ -170,13 +206,16 @@ def _get_key_id(currentKey):
...
-        fiscalyear = FiscalYear(data['fiscalyear'])
+        fiscalyear = (FiscalYear(data['fiscalyear']) if data.get('fiscalyear')
+            else None)
         start_period = None
         if data['start_period']:
             start_period = Period(data['start_period'])
         end_period = None
         if data['end_period']:
             end_period = Period(data['end_period'])
+        start_date = data['start_date'] if data.get('start_date') else None
```
Comment (bernatnan) [inline general_ledger/general_ledger.py]: Colud be:
start_date = data.get('start_date', None)

File: general_ledger/general_ledger.py [hunk 2]
```diff
@@ -170,13 +206,16 @@ def _get_key_id(currentKey):
...
+        end_date = data['end_date'] if data .get('end_date') else None
```
Comment (bernatnan) [inline general_ledger/general_ledger.py]: Colud be:
end_date = data.get('start_date', None)
And we solve the space between "data" and ".get"

File: general_ledger/general_ledger.py [hunk 3]
```diff
@@ -202,15 +241,24 @@ def _get_key_id(currentKey):
...
         parameters['company_vat'] = (company.party.tax_identifier
             and company.party.tax_identifier.code) or ''
         parameters['start_period'] = start_period and start_period or ''
         parameters['end_period'] = end_period and end_period or ''
-        parameters['fiscal_year'] = fiscalyear.rec_name
+        parameters['start_date'] = (start_date.strftime('%d/%m/%Y')
+            if start_date else '')
+        parameters['end_date'] = (end_date.strftime('%d/%m/%Y')
+            if end_date else '')
+        parameters['fiscal_year'] = fiscalyear.rec_name if fiscalyear else None
```
Comment (bernatnan) [inline general_ledger/general_ledger.py]: For "parameters" could be better use an empty string '', not a None.

Comment (raimonesteve) [reply general_ledger/general_ledger.py]: Is jinja, not jasper

```
{% if fiscal_year %} .... {% endif %}
```

not equal to:

```
{% if fiscal_year == '' %} .... {% endif %}
```

File: general_ledger/general_ledger.py (hunk 1 above)
Comment (raimonesteve) [reply general_ledger/general_ledger.py]: 6c876e1da733016c9605e244196422d4248e840e

File: general_ledger/general_ledger.py (hunk 3 above)
Comment (bernatnan) [reply general_ledger/general_ledger.py]: I know it's jinja. But I think for strings is better use:

parameters['fiscal_year'] = fiscalyear.rec_name if fiscalyear else ''

If it's necessary to check "fiscal_year" has information, you could do in jinja in various ways too. For example:

{% if fiscal_year|length %}

or

{% if fiscal_year != '' %}

Comment (raimonesteve) [reply general_ledger/general_ledger.py]: 472821e327fd1fdf067d3be78c3fbcdf7448bed2

**Review by bernatnan — APPROVED** [review]

**Pull request merged**